src/main/java/com/imsweb/x12/Separators.java

```diff
@@ -18,6 +22,7 @@ public Separators() {
         _segment = '~';
         _element = '*';
         _composite = ':';
+        setSeparatorPatterns(_segment, _element, _composite);
```

Comment:
Also, why does `setSeparatorPatterns` even exist?  Just call `setSegment`, `setElement` and `setCompositeElement` instead of directly assigning them in the constructor.  That way the class members get set and the Patterns are compiled.  It doesn't make sense to compile the patterns in two different places.